darwin.mk

```diff
@@ -0,0 +1,18 @@
+define build_darwin_lifecycle
```

Comment:
Is there anything darwin-specific about this script? Could it just be `build_lifecycle`?

Reply:
I'd lean toward a lifecycle.mk and launcher.mk in this case, since we know that we'll just be setting different env vars for different os/arch targets. But yes let's go ahead and update the other targets as a separate effort.